### PR TITLE
Fixed a few details + zig and c interoperability

### PR DESCRIPTION
**Pull request by Pismice**

### CODE REVIEW

File: docs/zig-and-c.org [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+const items = [_]u16 { 1, 4, 0, 1 };
+var sum: u16 = 0;
+for (items) |value| {
+    sum += value;
+}
+std.debug.print("Sum: {}\n", .{sum});
+#+end_src
+
+#+begin_src zig :imports '(std) :main 'yes :testsuite 'no
+var i: usize = 0;
```
Comment (samuel-schlaeppy) [inline docs/zig-and-c.org]: Now we can loop in zig with (V. 0.12):
```zig
var sum3 : usize = 0;
for (0..5) |i| {
    sum3 += i;
 }
```
https://ziglang.org/documentation/0.12.0/#for

File: docs/zig-and-c.org [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+#+begin_src zig :imports '(std) :main 'yes :testsuite 'no
+const items = [_]u16 { 1, 4, 0, 1 };
+var sum: u16 = 0;
+for (items) |value| {
+    sum += value;
+}
+std.debug.print("Sum: {}\n", .{sum});
+#+end_src
+
+#+begin_src zig :imports '(std) :main 'yes :testsuite 'no
```
Comment (samuel-schlaeppy) [inline docs/zig-and-c.org:39]: You don't need to put `:testsuite 'no`, it by default to `no`

File: docs/zig-and-c.org [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+pub fn main() !void {
+    var i: usize = 0;
+    while (i < 100000000) : (i += 1) {
+        _ = add(3, 7);
+    }
+    std.debug.print("done\n", .{});
+}
+#+end_src
+Result: ~0.38sec
```
Comment (samuel-schlaeppy) [inline docs/zig-and-c.org:183]: You run the code only once ? Maybe use hyperfine : https://github.com/sharkdp/hyperfine

File: docs/zig-and-c.org [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+Result: ~50sec
+
+TODO faire un joli graphique quand jaurai d autres languages
+TODO idk if it would be possible to compile my c library and then use it in my python code, and all of that all in emacs, that would be cool :)
+TODO do with other languages aswell
+
+***** Conclusion
+First thing that we notice immediately is how much faster the Zig code is compared to the Python code. This is not surprising since Zig is a compiled language and Python is an interpreted language.
+
+The second interesting thing is that the 2 Zig codes dont vary that much (if they even do) compared to the 2 python codes which have a 5x ratio. This is interesting because it shows that the overhead of calling a C function from Zig is not that big (in fact it is even not existent since all the compiler does is translating the C code to Zig code at compilation time).
```
Comment (samuel-schlaeppy) [inline docs/zig-and-c.org:239]: Was it the interoperability between Python and C that made the code slow or was it just the Python interpreter that couldn't optimise?

File: docs/zig-and-c.org (hunk 1 above)
Comment (Pismice) [reply docs/zig-and-c.org]: fixed it :)

File: docs/zig-and-c.org (hunk 2 above)
Comment (Pismice) [reply docs/zig-and-c.org:39]: Thats cool to know, how can I use it though ? I have just been copying the way you were doing but I dont have the updates you did to be able to run zig test

File: docs/zig-and-c.org (hunk 3 above)
Comment (Pismice) [reply docs/zig-and-c.org:183]: no, ran it multiple times
im going to use it hyperfine next time :)

File: docs/zig-and-c.org (hunk 4 above)
Comment (Pismice) [reply docs/zig-and-c.org:239]: very good question i added a TODO so that I can dig into it later :)

**Pull request merged**